static/js/message_edit.js

```diff
@@ -593,6 +595,30 @@ export function start(row, edit_box_open_callback) {
     });
 }
 
+export function handle_resolve_topic(recipient_row) {
+    const message_id = rows.id_for_recipient_row(recipient_row);
+    const message = message_lists.current.get(message_id);
+    const old_topic_name = message.topic;
+    let new_topic_name;
+    if (old_topic_name.startsWith(RESOLVED_TOPIC_PREFIX)) {
+        new_topic_name = _.trimStart(old_topic_name, RESOLVED_TOPIC_PREFIX);
+    } else {
+        new_topic_name = RESOLVED_TOPIC_PREFIX + old_topic_name;
+    }
+
+    const request = {
+        propagate_mode: "change_all",
+        topic: new_topic_name,
+        send_notification_to_old_thread: false,
+        send_notification_to_new_thread: true,
+    };
+
+    channel.patch({
+        url: "/json/messages/" + message.id,
+        data: request,
+    });
+}
```

Comment:
This is great, but we probably want to do the refactoring to make this share code with the `stream_popovers.js` logic.
I think structurally, if this function accepted a `message_id` and a topic, we could have the `stream_popover.js` code call it?

Reply:
(Maybe it makes sense to do this as a series of preparatory refactoring commits).